File: web/src/main/java/ru/protei/portal/ui/web/client/model/event/EventBusEventTest.java
package ru.protei.portal.ui.web.client.model.event;

import jsinterop.annotations.JsOverlay;

public class EventBusEventTest extends EventBusEvent {
    protected EventBusEventTest() {}
    public static String type = "@app-portal/test-event";

    public static native EventBusEventTest create(String text) /*-{
        return {
            type: @ru.protei.portal.ui.web.client.model.event.EventBusEventTest::type,
            source: @ru.protei.portal.ui.web.client.model.event.EventBusSource::SourcePortalUiGwt,
            payload: {
                text: text
            }
        };
    }-*/;

    @JsOverlay public final native String getText() /*-{ return this.payload.text; }-*/;
}

File: webts/packages/common/common/src/js/model/eventbus/event/EventBusEventTest.ts
import { EventBusEvent, EventSource } from "@protei-libs/eventbus"
import { EventBusSourcePortalUi } from "../source/EventBusSource"

export const EventBusEventTestType = "@app-portal/test-event"

export type EventBusEventTest = ReturnType<typeof eventBusEventTest>

export type EventBusEventTestPayload = {
  text: string
}

export const eventBusEventTest = (
  payload: EventBusEventTestPayload,
  options?: { source?: EventSource },
): EventBusEvent<EventBusEventTestPayload> =>
  <const>{
    type: EventBusEventTestType,
    source: options?.source ?? EventBusSourcePortalUi,
    payload: payload,
  }
